docs(DockManager): add dockmanager documentation (#2742)
telerik/blazor-docs: components/dockmanager/overview.md

---
title: Overview
page_title: DockManager Overview
description: Overview of the DockManager for Blazor.
slug: dockmanager-overview
tags: telerik,blazor,dockmanager,overview
published: True
position: 0
---

# Blazor DockManager Overview

The <a href="https://www.telerik.com/blazor-ui/dock-manager" target="_blank">Blazor DockManager component</a> is a versatile tool that enables users to manage and organize multiple panes within a single container. It supports features like docking, undocking, resizing, and repositioning, offering a flexible and customizable layout.

> >tip The DockManager is best suited for desktop-like interfaces and applications designed for larger screens, where users can take full advantage of its advanced layout management capabilities.

Reviewer: Using a highlighted tip implies that we are defensive about something. Better append this sentence to the introduction simply to manage customer expectations.

---
title: Overview
page_title: DockManager Overview
description: Overview of the DockManager for Blazor.
slug: dockmanager-overview
tags: telerik,blazor,dockmanager,overview
published: True
position: 0
---

# Blazor DockManager Overview

The <a href="https://www.telerik.com/blazor-ui/dock-manager" target="_blank">Blazor DockManager component</a> is a versatile tool that enables users to manage and organize multiple panes within a single container. It supports features like docking, undocking, resizing, and repositioning, offering a flexible and customizable layout.

## Creating Blazor DockManager

1. Add the `TelerikDockManager` tag.
2. Use `<DockManagerPanes>` to structure the main docked layout.
3. Within <DockManagerPanes>, add:
    * `<DockManagerContentPane>` for standalone panes.

>     * `<DockManagerSplitPane>` to create resizable sections containing multiple panes.

Reviewer: Is it "resizable sections with multiple panes" or is it "sections with multiple resizable panes" ?

    * `<DockManagerTabGroupPane>` to enable tabbed panes.
4. Define `HeaderTemplate` inside each pane to set the headers text.

> 5. Populate the `Content` section of each pane with the desired UI elements.

Reviewer: If you use tags, then use tags for the `HeaderTemplate` and `Content` too. Otherwise it may look like these are used differently.

6. Optionally, `<DockManagerFloatingPanes>` to create panes that can float outside the main dock layout.

>caption Telerik Blazor DockManager

````RAZOR

> <TelerikDockManager>

Reviewer: Consider the following improvements:
* Make the example simpler by removing the repetitive panes and long content. It may be overwhelming for first-time users.
* Make the configuration suitable to fit in small space in the docs.
* Set some pane parameters, which are most likely to be used, such as `Size` and `Orientation`.

For example:

```razor
<TelerikDockManager Height="90vh">
    <DockManagerPanes>

        <DockManagerSplitPane Orientation="@DockManagerPaneOrientation.Vertical"
                              Size="40%">
            <Panes>

                <DockManagerContentPane Size="55%">
                    <HeaderTemplate>
                        <strong>Pane 1.1</strong>
                    </HeaderTemplate>
                    <Content>
                        First Content Pane in Split configuration
                    </Content>
                </DockManagerContentPane>

                <DockManagerContentPane>
                    <HeaderTemplate>
                        <strong>Pane 1.2</strong>
                    </HeaderTemplate>
                    <Content>
                        Second Content Pane in Split configuration
                    </Content>
                </DockManagerContentPane>

            </Panes>
        </DockManagerSplitPane>

        <DockManagerTabGroupPane>
            <Panes>

                <DockManagerContentPane>
                    <HeaderTemplate>
                        <strong>Tab 2.1</strong>
                    </HeaderTemplate>
                    <Content>
                        First Tab Content
                    </Content>
                </DockManagerContentPane>

                <DockManagerContentPane>
                    <HeaderTemplate>
                        <strong>Tab 2.2</strong>
                    </HeaderTemplate>
                    <Content>
                        Second Tab Content
                    </Content>
                </DockManagerContentPane>

            </Panes>
        </DockManagerTabGroupPane>
    </DockManagerPanes>

    <DockManagerFloatingPanes>
        <DockManagerSplitPane>
            <Panes>

                <DockManagerContentPane>
                    <HeaderTemplate>
                        <strong>Floating Pane</strong>
                    </HeaderTemplate>
                    <Content>
                        Floating Pane Content
                    </Content>
                </DockManagerContentPane>

            </Panes>
        </DockManagerSplitPane>
    </DockManagerFloatingPanes>
</TelerikDockManager>
```

                    </Panes>
                </DockManagerTabGroupPane>

                <DockManagerContentPane>
                    <HeaderTemplate>
                        <strong>Notifications</strong>
                    </HeaderTemplate>
                    <Content>
                        <p>New messages, mentions, and important updates.</p>
                    </Content>
                </DockManagerContentPane>

            </Panes>
        </DockManagerSplitPane>
    </DockManagerPanes>

    <DockManagerFloatingPanes>
        <DockManagerSplitPane>
            <Panes>

                <DockManagerContentPane>
                    <HeaderTemplate>
                        <strong>Team Chat</strong>
                    </HeaderTemplate>
                    <Content>
                        <p>Live conversation with team members.</p>
                        <TelerikTextBox @bind-Value="@ChatMessage"></TelerikTextBox>
                    </Content>
                </DockManagerContentPane>

            </Panes>
        </DockManagerSplitPane>
    </DockManagerFloatingPanes>
</TelerikDockManager>

@code {
    private string ChatMessage { get; set; }
}
````

## State

> The [Dock Manager allows getting and setting its state](slug://dockmanager-state). The DockManager state contains information about the panes hierarchy, a collection of all the floating panes, and all the DockManager configurations, such as its orientation.

Reviewer: ```suggestion
The [Dock Manager allows getting and setting its state](slug://dockmanager-state). The DockManager state contains information about the pane hierarchy, floating panes, current pane settings, and the DockManager configuration, such as its orientation.
```

> ## Dock Types

Reviewer: "Dock Types" implies that "dock" is a noun, while it's a verb. Consider "Docking Types", similar to the heading in the linked article.

The DockManager exposes the ability to dock globally or within other panes. [Read more about the available DockManager dock types...](slug://dockmanager-dock-types)

## Pane Types

The DockManager exposes the ability to configure different pane types. [Read more about the DockManager pane types...](slug://dockmanager-pane-types)

## Events

The Dock Manager fires [events when the user changes the panes layout](slug://dockmanager-events). This allows custom logic execution, refreshing of nested components and saving the [DockManager state](slug://dockmanager-state) for later restore.

## DockManager Parameters

> The following table lists the Dock Manager parameters. Also check the [DockManager API Reference](slug://Telerik.Blazor.Components.TelerikTileLayout) for a full list of all properties, methods and events.

Reviewer: Wrong link

### DockManagerTabGroupPane Parameters

| Parameter | Type and Default&nbsp;Value | Description |
| --- | --- | --- |
| `AllowEmpty` | `bool` <br /> (`false`) | Determines whether an empty space is left when all tabs are removed (unpinned, closed, etc.), allowing you to drop content panes and create a new tab. |
| `Id` | `string` <br /> (`Guid`) | The id of the pane. |
| `SelectedPaneId` | `int` | The `id` of the initially selected tab pane. |
| `Size` | `string` | Determines the size of the splitter pane. |
| `Visible` | `bool` <br /> (`true`) | Determines whether the tab/pane is rendered. |

## DockManager Reference

Use the component reference to execute methods and [get or set the DockManager state](slug://dockmanager-state).

| Method | Description |
| --- | --- |
| `GetState` | Returns the current state of the Dock Manager as a [`DockManagerState` object](slug://Telerik.Blazor.Components.DockManagerState). |
| `Refresh` | Use the method to programmatically re-render the component.  |
| `SetState` | Applies the provided `DockManagerState` argument as a new state of the Dock Manager. |

<div class="skip-repl"></div>

````RAZOR

> <TelerikDockManager @ref="@DockRef" />
> 
> <TelerikButton OnClick="@GetDockManagerState">Get DockManager State</TelerikButton>
> 
> @code{
>     private TelerikDockManager DockRef { get; set; }
> 
>     private async Task GetDockManagerState()
>     {
>         var dockState = DockRef.GetState();
>     }
> }

Reviewer: ```suggestion
<TelerikDockManager @ref="@DockManagerRef" />

<TelerikButton OnClick="@GetDockManagerState">Get DockManager State</TelerikButton>

@code{
    private TelerikDockManager? DockManagerRef { get; set; }

    private void GetDockManagerState()
    {
        var dockState = DockManagerRef?.GetState();
    }
}
```
